### PR TITLE
fs: consistently use fsbridge:: for ifstream / ofstream

### PR DESCRIPTION
**Pull request by fanquake**
These changes are part of #20744, but are also ok to do now, and reduce the diff in that PR. See commit messages for details. Revived from #23857.

### CODE REVIEW

**Comment by hebasto** [comment]
Concept ACK.

File: src/bitcoin-tx.cpp [hunk 1]
```diff
@@ -158,7 +159,7 @@ static void RegisterLoad(const std::string& strInput)
     std::string key = strInput.substr(0, pos);
     std::string filename = strInput.substr(pos + 1, std::string::npos);
 
-    FILE *f = fopen(filename.c_str(), "r");
+    FILE *f = fsbridge::fopen(filename.c_str(), "r");
```
Comment (MarcoFalke) [inline src/bitcoin-tx.cpp:162]: unrelated question for follow-up: Why is this safe? Shouldn't this use PathFromString?

**Review by maflcko — APPROVED** [review]
ACK 5e8975e2694c3178ae73deb28986e1fb5466147e 🏕

<details><summary>Show signature</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

ACK 5e8975e2694c3178ae73deb28986e1fb5466147e 🏕
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUg91Av+IL93ayKaf1O3lEaZkky3cz+mduqUCE8hGE6UQo9NSM47OvHEeTRPVK0H
DHTpZidWXOJbzuBvKhs7YLOm4xgYN5Wx2/UDtFHIMFTQynsxZjPYDILWPNXjUBma
pwNS59KsWZO1iZNqBaG+gHUA2zGIbVDLCP+wY/HbeTDIOr+Sr27Y3lxha2AUDYKr
O1J+Hxre7S9nDSp+x5i9XcL6qpw0xs8nzqweRYd1gBUGkEj1FzpcYBgh2QCbuOqD
idxw9dA8f3m1Y+1P/kpSKw3Kzvc/Ce18gOmCzUMKpXEfwfLsbPOXG468xviJpc5Y
1EaFsrZHUMOpz6VRFZ+bl/xfH3vCdCMZY9XNyV4HzG8Rnz7JHzJS4qXmx8eILKK0
V9uc7sseLDJ9ttX1Vy5/hyWDOnSnu8YPpt3f4gPeGA79+sjBMOMG5451hSzazNP4
4yQs8uUVEWH/j97XQQFdD6ldFQtrD2dU4zb3mxnA7V8xPZmMjaspJE4rCZveytas
ty+5S2DM
=nQnW
-----END PGP SIGNATURE-----
```


</details>

**Comment by laanwj** [comment]
It's good to be consistent with this.
Code review ACK 5e8975e2694c3178ae73deb28986e1fb5466147e

**Pull request merged**